src/scripts/index.ts

```diff
@@ -0,0 +1,75 @@
+import { Map } from "leaflet";
+import { getWeatherData } from "./api.ts";
+import { setDOM } from "./dom.ts";
+import { initilizeMap, updateMap, map_listeners } from "./map.ts";
+
+// GLOBAL VARIABLES
+const map: Map = initilizeMap();
+
+// EVENT LISTENERSSS
+window.addEventListener("load", (event) => {
+  getCoords("Johannesburg");
+});
+
+// map listeners
+const map_div: HTMLElement | null = document.getElementById("map_container");
+const seven_day: HTMLElement | null =
+  document.getElementById("seven_day_forecast");
+const open_map: HTMLElement | null =
+  document.getElementById("map_button_float");
+
+if (map_div && seven_day && open_map) {
+  map_listeners(map, map_div, seven_day, open_map);
+}
```

Comment:
As above, I would rather not let the types possibly be null.
```ts
const map_div = document.getElementById("map_container");
const seven_day = document.getElementById("seven_day_forecast");
const open_map = document.getElementById("map_button_float");

if (!map_div || seven_day || open_map) {
  //Handle what happens when these are null. For example, throw an error (this would require a try/catch.
}

mapListeners(map, map_div, seven_day, open_map);
```